ArdeleanuStefan_Task5/include/DynamicArray.hpp

```diff
@@ -0,0 +1,170 @@
+#pragma once
+
+#include <cstring>
+#include <iostream>
+#include <string>
+#include <vector>
+
+template <typename T>
+class DynamicArray
+{
+private:
+    T *dynamicArray;
+    size_t m_size;
+
+public:
+    DynamicArray() : m_size{0}, dynamicArray{nullptr}
+    {
+    }
+
+    DynamicArray(const std::vector<T> &inputs) : DynamicArray()
+    {
+        for (auto input : inputs)
+            addEntry(input);
+    }
+
+    DynamicArray(DynamicArray<T> &&obj) : m_size{obj.m_size}, dynamicArray{obj.dynamicArray}
+    {
+        obj.m_size = 0;
+        obj.dynamicArray = nullptr;
+    }
+
+    DynamicArray<T> &operator=(DynamicArray<T> &&obj)
+    {
+        if (this != &obj)
+        {
+            delete[] dynamicArray;
+
+            dynamicArray = obj.dynamicArray;
+            m_size = obj.m_size;
+
+            obj.dynamicArray = nullptr;
+            obj.m_size = 0;
+        }
+        return *this;
+    }
+
+    size_t get_size() const
+    {
+        return m_size;
+    }
+
+    void addEntry(const T &input)
+    {
+        T *temp = new T[m_size + 1];
+        for (size_t i = 0; i < m_size; ++i)
+            temp[i] = dynamicArray[i];
+        temp[m_size] = input;
+        ++m_size;
+        delete[] dynamicArray;
+        dynamicArray = temp;
+    }
+
+    bool deleteEntry(const T &input)
+    {
+        bool existString = false;
+        T *temp = nullptr;
+        for (size_t i = 0; i < m_size; i++)
+        {
+            if (dynamicArray[i] == input)
+            {
+                existString = true;
+                break;
+            }
+        }
+
+        if (existString)
+        {
+            if (m_size > 1)
+            {
+                temp = new T[m_size - 1];
+                int index = 0;
+                for (size_t i = 0; i < m_size; i++)
+                {
+                    if (dynamicArray[i] == input)
+                    {
+                        continue;
+                    }
+                    temp[index] = dynamicArray[i];
+                    index++;
+                }
+            }
+            delete[] dynamicArray;
+            dynamicArray = temp;
+            --m_size;
+        }
+        return existString;
+    }
+
+    T *getEntry(const int &index) const
+    {
+
+        try
+        {
+            if (index < 0 || index >= m_size)
+            {
+                throw std::out_of_range("INDEX IS OUT OF RANGE");
+            }
+            else
+            {
+                return &dynamicArray[index];
+            }
+        }
+        catch (const std::out_of_range &e)
```

Comment:
I think you should catch the exceptions outside the method.
You also should specify that the method will throw an exception.